node-tests/setup-middleware-test.js

```diff
@@ -0,0 +1,59 @@
+const QUnit = require('qunit');
+const fetch = require('node-fetch');
+const fs = require('fs');
+const tmp = require('tmp');
+const express = require('express');
+const readJSONSync = require('fs-extra').readJSONSync;
+const setupMiddleware = require('../setup-middleware');
+const violationsFixture = require('./fixtures/violations');
+
+function createTmpDir() {
+  return fs.realpathSync(tmp.dirSync({ unsafeCleanup: true }).name);
+}
+
+function buildResult(axeResults) {
+  let { module, testName } = QUnit.config.current;
+
+  return {
+    moduleName: module.name,
+    testName,
+    helperName: 'visit',
+    stack: 'STACK',
+    axeResults,
+  };
+}
```

Comment:
I think this is still not quite right (we don't want the JSON file to be one per a11yAudit invocation). I'm happy to iterate in separate PRs, but lets make an issue to chat through the goal/intent here (and the final format).

Reply:
Are you saying the test isn't right? The test doesn't mimic exactly how this will be wired up IRL, but merely tests the emitting and gathering of the data. 

Can you be more specific?